esp/services/WsDeploy/WsDeployService.cpp

```diff
@@ -974,6 +974,17 @@ bool CWsDeployFileInfo::saveSetting(IEspContext &context, IEspSaveSettingRequest
       const char* pszOnChange = pSetting->queryProp("@onChange");
       const char* pszViewType = pSetting->queryProp("@viewType");
 
+      //
+      // Trim trailing blanks from ECL Agent queue name
+      if (!strcmp(pszAttrName, "wuQueueName") && strlen(pszNewValue))
+      {
+        char *pValue = const_cast<char *>(pszNewValue);
+        char *end = pValue + strlen(pValue) - 1;
+        while(end > pValue && isspace((unsigned char)*end))
+          end--;
+        *(end+1) = 0;
+      }
```

Comment:
If you use StringBuffer to store pszNewValue, you may use trimRight() and make the code much clear.